### PR TITLE
Toggle Switch - Phase 2 docs

### PR DESCRIPTION
**Pull request by Blackbaud-DeniseCisneros**
Phase 2 of Toggle Switch docs. Pending demo and code examples

### CODE REVIEW

File: src/app/index.html [hunk 1]
```diff
@@ -7,6 +7,9 @@ <h2>
   <li>
     <a skyAppLink="/docs/radio">Radio</a>
   </li>
+  <li>
+    <a skyAppLink="/docs/toggle-switch">Toggle Switch</a>
```
Comment (blackbaud-johnly) [inline src/app/index.html:11]: Probably doesn't matter much, but `switch` should be lowercase to follow our sentence-style capitalization pattern.

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+          </ul>
+        </sky-docs-design-guideline>
+      </sky-docs-demo-page-section>
+    </sky-docs-design-guidelines>
+
+    <sky-docs-code-examples>
+      <sky-docs-code-example heading="Toggle"
+        sourceCodePath="src/app/public/plugin-resources/code-examples/toggle-switch/basic"></sky-docs-code-example>
+    </sky-docs-code-examples>
+
```
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html:194]: Also, it looks like we only have one code example. In the old demo, we had an example of a reactive form and a template-driven form. Should we have both here as well? Might be worth asking Steve about that.

[the remark this answers is not in the document]
Comment (Blackbaud-SomaliaJamall) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html:194]: I copied the code example from the specs that @Blackbaud-AdamFunderburk created, I know there's a few of them that are a bit different from the old code demos. I assumed the new ones had already been vetted, perhaps adam can weigh in?

**Comment by Blackbaud-SomaliaJamall** [comment]
I'll work today to see if I can figure out why the checks are failing. It's strange, that check is passing on my local machine.

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html:194]: The template-driven form vs. reactive form difference might not matter in phase 2 docs. We can ask Adam at the docs check-in on Tuesday.

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2](https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2)

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            caption="Do use toggles to set binary states on multiple items."
+            captionType="success"
+            imageSource="~/assets/img/guidelines/toggle-switch/toggle-usage-2.png"
+          >
+          </sky-docs-design-guideline-thumbnail>
+        </sky-docs-design-guideline>
+
+        <sky-docs-design-guideline
+          heading="Don't use when"
+          headingType="danger"
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html]: ✔️

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+          <sky-docs-design-guideline-thumbnail
+            caption="Do use toggles to set binary states on multiple items."
+            captionType="success"
+            imageSource="~/assets/img/guidelines/toggle-switch/toggle-usage-2.png"
+          >
+          </sky-docs-design-guideline-thumbnail>
+        </sky-docs-design-guideline>
+
+        <sky-docs-design-guideline
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html:60]: ✔️

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+      </sky-docs-demo-page-section>
+
+      <sky-docs-demo-page-section
+        heading="Options"
+      >
+        <sky-docs-design-guideline
+          heading="Text label"
+        >
+          <p>
+            When toggles are in grid columns, the text label is not required if you use a column heading. Use text labels for all other states.
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html]: ✔️

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html:197]: ✔️

File: src/app/index.html (hunk 1 above)
Comment (Blackbaud-DeniseCisneros) [reply src/app/index.html]: ✔️

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+          </p>
+          <p>
+            If the scenario allows for toggles to become disabled, always include a text label on each toggle.
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html]: I went ahead and just made the change, but I can easily just put it back to what it was if they disagree.

@Blackbaud-AdamFunderburk or @Blackbaud-ToddRoberts Any strong thoughts on this change?

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+          >
+          </sky-docs-design-guideline-thumbnail>
+        </sky-docs-design-guideline>
+
+        <sky-docs-design-guideline
+          heading="Don't use when"
+          headingType="danger"
+        >
+          <p>
+            Don’t use toggles when the options have significant consequences that the user needs to confirm before changes take effect. Use a workflow where users must confirm the changes instead.
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html]: ✔️

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+        </sky-docs-design-guideline>
+      </sky-docs-demo-page-section>
+
+      <sky-docs-demo-page-section
+        heading="Anatomy"
+      >
+        <sky-docs-anatomy
+          imagePath="~/assets/img/guidelines/toggle-switch/toggle-anatomy.png"
+        >
+          <sky-docs-anatomy-item>Toggle Body</sky-docs-anatomy-item>
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html]: ✔️

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+  moduleName="SkyToggleSwitchModule"
+  moduleSourceCodePath="src/app/public/modules/toggle-switch/"
+  pageTitle="Toggle Switch"
+>
+
+  <sky-docs-demo-page-summary>
+    The toggle switch component renders a SKY UX-themed switch for values that can either be "on" or "off." The change takes effect immediately.
+  </sky-docs-demo-page-summary>
+
+  <sky-docs-demo>
```
[the remark this answers is not in the document]
Comment (Blackbaud-DeniseCisneros) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html:11]: @Blackbaud-SomaliaJamall Would you be able to help with that functionality? I see that inline delete has an `onResetClick` method for the demo items. I'm assuming we need something similar.

File: package.json [hunk 10]
```diff
@@ -28,7 +28,9 @@
     "@skyux/i18n": "^3.5.0",
     "@skyux/indicators": "^3.0.0"
   },
-  "dependencies": {},
+  "dependencies": {
+    "ajv": "^6.12.0"
```
Comment (Blackbaud-SteveBrush) [inline package.json:32]: Mind moving this to `devDependencies`?

Comment (Blackbaud-DeniseCisneros) [reply package.json]: Done!

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2](https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2)

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2](https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2)

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html]: @Blackbaud-AdamFunderburk and @Blackbaud-ToddRoberts, just checking back in to see if you guys are OK with this change.

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html]: @Blackbaud-AdamFunderburk and @Blackbaud-ToddRoberts, just checking in to see if you guys are OK with this change.

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html (hunk 9 above)
[the remark this answers is not in the document]
Comment (blackbaud-johnly) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html:11]: I just played around with the demo, and the reset option now reverts the selection back to it's original state. Thanks!

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2](https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2)

File: src/app/docs/toggle-switch/toggle-switch-docs.component.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (Blackbaud-ToddRoberts) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html]: I think the new wording is generally good. @Blackbaud-AdamFunderburk I don't remember why we decided that disabled toggles always need labels, even in grids, but I do remember that it was intentional. Should we add that back in?

[the remark this answers is not in the document]
Comment (Blackbaud-AdamFunderburk) [reply src/app/docs/toggle-switch/toggle-switch-docs.component.html]: I think it's important to maintain the point that labels should be used if the switches can be disabled.

"If toggle switches can be disabled, include individual text labels to reflect their state."

I understand how tangled that makes the guidelines around the text label, and I'm not sure quite how to optimize them.

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2](https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2)

**Comment by Blackbaud-BobbyEarl** [comment]
Preview build at [https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2](https://host.nxt.blackbaud.com/skyux-forms?_sp.spa=pr-toggleswitch-phase2)

**Pull request merged**